atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CQLKeyValueService.java

```diff
@@ -1349,6 +1340,69 @@ public Void call() throws Exception {
 
     @Override
     public void compactInternally(String tableName) {
-        // nothing to do, yet...
+        Preconditions.checkArgument(!Strings.isNullOrEmpty(tableName), "tableName:[%s] should not be null or empty", tableName);
+        long compactionTimeoutSeconds = config.jmxCompactionTimeoutSeconds();
+        try {
+            alterGcGraceSeconds(tableName, 0);
+            alterTombstoneThreshold(tableName, 0.0f);
+            compactionManager.forceTableCompaction(compactionTimeoutSeconds, config.keyspace(), tableName);
+        } catch (TimeoutException e) {
+            log.error("Compaction could not finish in {} seconds! {}", compactionTimeoutSeconds, e.getMessage());
+            log.error(compactionManager.getPendingCompactionStatus());
+        } finally {
+            alterGcGraceSeconds(tableName, CassandraConstants.GC_GRACE_SECONDS);
+            alterTombstoneThreshold(tableName, CassandraConstants.TOMBSTONE_THRESHOLD_RATIO);
+        }
+    }
+
+    private void alterGcGraceSeconds(final String tableName, int gcGraceSeconds) {
```

Comment:
I'm taking it at face value that these are correct since they're likely clean ports from the internal stuff.